Adding _update_packages recipe

mohanasudhan opened the pull request:
We want to move package update from cloudformation
template bootstrap action as it can create different bits over time.
Adding separate recipe and attaching only during kitchen and packer build

This commit reverts packer changes https://github.com/awslabs/cfncluster-cookbook/pull/44

Signed-off-by: Mohan Gandhi <mohgan@amazon.com>

Started testing this changes locally. will update the results soon

Packer - alinux: picks up _update_packages
1510098033,,ui,message,    amazon-ebs: Recipe: cfncluster::_update_packages
1510098033,,ui,message,    amazon-ebs: * execute[yum-update] action run[2017-11-07T23:40:33+00:00] INFO: Processing execute[yum-update] action run (cfncluster::_update_packages line 18)
1510098033,,ui,message,    amazon-ebs:
1510098033,,ui,message,    amazon-ebs: [execute] Loaded plugins: priorities%!(PACKER_COMMA) update-motd%!(PACKER_COMMA) upgrade-helper
1510098033,,ui,message,    amazon-ebs: No packages marked for update
1510098033,,ui,message,    amazon-ebs: [2017-11-07T23:40:33+00:00] INFO: execute[yum-update] ran successfully
1510098033,,ui,message,    amazon-ebs: - execute yum -y update

kitchen also picks up _update_packages

Converging 124 resources
       Recipe: cfncluster::_update_packages
         * execute[yum-update] action run[2017-11-07T23:53:05+00:00] INFO: Processing execute[yum-update] action run (cfncluster::_update_packages line 18)
       
           [execute] Loaded plugins: priorities, update-motd, upgrade-helper
              No packages marked for update
       [2017-11-07T23:53:05+00:00] INFO: execute[yum-update] ran successfully
           - execute yum -y update

bwbarrett commented on recipes/_update_packages.rb:
```diff
@@ -0,0 +1,26 @@
+case node['platform_family']
+when 'rhel'
+  execute 'yum-update' do
+    command "yum -y update"
```
We've accrued a bunch of old kernels in the CentOS AMIs as a result of all the upgrades along the way.  We should add a `package-cleanup --oldkernels --count=1` here to remove the older kernels.

mohanasudhan replied:
I can add this line in recipe. You have the same in the new packer script. Adding in recipe doesn't harm us, let me know if this comment still holds as of now.

bwbarrett replied:
it's actually `package-cleanup -y --oldkernels --count=1`, and yeah, we should run it more regularly.

mohanasudhan commented:
Addressed comments:

1. two files for kitchen -- dougal added the rationale behind it (make sense to me)  https://github.com/awslabs/cfncluster/issues/254
2. moved to two commits
3. updated commit messages

bwbarrett commented on recipes/_update_packages.rb:
```diff
@@ -0,0 +1,26 @@
+case node['platform_family']
+when 'rhel'
+  execute 'yum-update' do
+    command "yum -y update && package-cleanup --oldkernels --count=1"
```
`package-cleanup -y --oldkernels --count=1`; without the `-y`, it'll prompt for input.

mohanasudhan replied:
ah, done now !
